crates/sui/src/client_commands.rs

```diff
@@ -3226,3 +2593,126 @@ pub async fn max_gas_budget(context: &mut WalletContext) -> Result<u64, anyhow::
         ),
     })
 }
+
+/// Dry run, execute, or serialize a transaction.
+///
+/// This basically extracts the logical code for each command that deals with dry run, executing,
+/// or serializing a transaction and puts it in a function to reduce code duplication.
+// TODO (stefan): Add gas_price option for all commands and remove it from this function
+pub(crate) async fn dry_run_or_execute_or_serialize(
+    signer: SuiAddress,
+    tx_kind: TransactionKind,
+    context: &mut WalletContext,
+    gas_payment: Option<Vec<ObjectID>>,
+    gas_price: Option<u64>,
+    opts_with_gas: Option<OptsWithGas>,
+    opts: Option<Opts>,
```

Comment:
Seems like instead of doing this, you want to have an `Option<Opts>` and a separate parameter for the optional gas params.